presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -400,4 +405,46 @@ public void setUGI(String userName)
     {
         client.set_ugi(userName, new ArrayList<>());
     }
+
+    @Override
+    public Optional<PrimaryKeysResponse> getPrimaryKey(String dbName, String tableName)
+            throws TException
+    {
+        PrimaryKeysRequest pkRequest = new PrimaryKeysRequest(dbName, tableName);
+        PrimaryKeysResponse pkResponse;
+
+        try {
+            pkResponse = client.get_primary_keys(pkRequest);
+        }
+        catch (TException e) {
+            // If we are talking to Hive version < 3 which doesn't support table constraints,
+            // then we will get an error like "Invalid Method name : get_primary_keys"
+            if (e.getMessage().contains("Invalid method name: 'get_primary_keys'")) {
```

Comment:
Do they have an error code? Checking error message feels shaky.

Reply:
Yes, changed it to use error code.